apps/meteor/ee/tests/unit/apps/livechat-enterprise/server/hooks/afterInquiryQueued.spec.ts

```diff
@@ -0,0 +1,120 @@
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+import moment from 'moment';
+import proxyquire from 'proxyquire';
+import sinon from 'sinon';
+
+const settingStub = {
+	watch: sinon.stub(),
+	get: sinon.stub(),
+};
+
+const callbackStub = {
+	add: sinon.stub(),
+	remove: sinon.stub(),
+	priority: { HIGH: 'high' },
+};
+
+const queueMonitorStub = {
+	scheduleInquiry: sinon.stub(),
+};
+
+const { afterInquiryQueued } = proxyquire
+	.noCallThru()
+	.load('../../../../../../app/livechat-enterprise/server/hooks/afterInquiryQueued.ts', {
+		'../../../../../app/settings/server': {
+			settings: settingStub,
+		},
+		'../../../../../lib/callbacks': {
+			callbacks: callbackStub,
+		},
+		'../lib/QueueInactivityMonitor': {
+			OmnichannelQueueInactivityMonitor: queueMonitorStub,
+		},
+		'../lib/logger': {
+			cbLogger: { debug: sinon.stub() },
+		},
+	});
+
+describe('hooks/afterInquiryQueued', () => {
+	beforeEach(() => {
+		callbackStub.add.resetHistory();
+		callbackStub.remove.resetHistory();
+		queueMonitorStub.scheduleInquiry.resetHistory();
+		settingStub.get.resetHistory();
+	});
+
+	it('should call settings.watch at first', () => {
+		expect(settingStub.watch.callCount).to.be.equal(1);
+	});
+
+	it('should call the callback on settings.watch with proper values', () => {
+		const func = settingStub.watch.getCall(0).args[1];
+
+		func(1);
+		expect(callbackStub.add.callCount).to.be.equal(1);
+
+		func(2);
+		expect(callbackStub.add.callCount).to.be.equal(2);
```

Comment:
This counts how many times the callback is being called, but the callback itself is being added only to the first call, right? So it's kind of useless, right?

Reply:
Not 100% useless, since we're checking it actually reached this point. Even when the original `callback.add` would do nothing, the code will call it anyways so this is a test of "when value is > 0, callbacks.add is called"